src/Features/Core/Portable/SplitOrMergeIfStatements/AbstractMergeIfStatementsCodeRefactoringProvider.cs

```diff
@@ -66,14 +66,22 @@ private async Task<Document> RefactorAsync(Document document, TextSpan upperIfOr
 
             var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
 
-            var upperIfOrElseIf = root.FindNode(upperIfOrElseIfSpan);
-            var lowerIfOrElseIf = root.FindNode(lowerIfOrElseIfSpan);
+            var upperIfOrElseIf = FindIfOrElseIf(upperIfOrElseIfSpan, ifGenerator, root);
+            var lowerIfOrElseIf = FindIfOrElseIf(lowerIfOrElseIfSpan, ifGenerator, root);
 
             Debug.Assert(ifGenerator.IsIfOrElseIf(upperIfOrElseIf));
             Debug.Assert(ifGenerator.IsIfOrElseIf(lowerIfOrElseIf));
 
             var newRoot = GetChangedRoot(document, root, upperIfOrElseIf, lowerIfOrElseIf);
             return document.WithSyntaxRoot(newRoot);
+
+            static SyntaxNode FindIfOrElseIf(TextSpan span, IIfLikeStatementGenerator ifGenerator, SyntaxNode root)
+            {
+                var innerMatch = root.FindNode(span, getInnermostNodeForTie: true);
+                return innerMatch?.FirstAncestorOrSelf<SyntaxNode, (TextSpan span, IIfLikeStatementGenerator ifGenerator)>(
+                    (node, arg) => arg.ifGenerator.IsIfOrElseIf(node) && node.Span == arg.span,
+                    (span, ifGenerator));
```

Comment:
isn't this in the fix?  i think needing to avoid allocation in this fashion is unnecessary.  just capture the values in a closure.  it will be a *lot* easier to read/understand.

Reply:
Changed to a closure